test/eventbridge_rule_with_sns_fanout/producer_stack/producer_stack.py

```diff
@@ -63,34 +82,69 @@ def __init__(self, scope: Construct, id: str, **kwargs) -> None:
         sns_permissions = iam.PolicyStatement()
         sns_permissions.add_any_principal()
         sns_permissions.add_actions(
-                        "SNS:Publish",
-                        "SNS:RemovePermission",
-                        "SNS:SetTopicAttributes",
-                        "SNS:DeleteTopic",
-                        "SNS:ListSubscriptionsByTopic",
-                        "SNS:GetTopicAttributes",
-                        "SNS:AddPermission",
-                        "SNS:Subscribe"
-                    )
+            "SNS:Publish",
+            "SNS:RemovePermission",
+            "SNS:SetTopicAttributes",
+            "SNS:DeleteTopic",
+            "SNS:ListSubscriptionsByTopic",
+            "SNS:GetTopicAttributes",
+            "SNS:AddPermission",
+            "SNS:Subscribe",
+        )
         sns_permissions.add_resources(topic.topic_arn)
-        sns_permissions.add_condition("StringEquals", {"AWS:SourceOwner": Aws.ACCOUNT_ID})
+        sns_permissions.add_condition(
+            "StringEquals", {"AWS:SourceOwner": Aws.ACCOUNT_ID}
+        )
         topic.add_to_resource_policy(sns_permissions)
 
         # Set up cross-account Subscription permissions for every onboarded language.
         subscribe_permissions = iam.PolicyStatement()
-        subscribe_permissions.add_arn_principal(f'arn:aws:iam::{Aws.ACCOUNT_ID}:root')
-        for id in account_ids:
-            subscribe_permissions.add_arn_principal(f'arn:aws:iam::{id}:root')
+        subscribe_permissions.add_arn_principal(f"arn:aws:iam::{Aws.ACCOUNT_ID}:root")
+        for id in all_language_account_ids:
+            subscribe_permissions.add_arn_principal(f"arn:aws:iam::{id}:root")
         subscribe_permissions.add_actions("SNS:Subscribe")
         subscribe_permissions.add_resources(topic.topic_arn)
         topic.add_to_resource_policy(subscribe_permissions)
 
         # Set up cross-account Publish permissions for every onboarded language.
         publish_permissions = iam.PolicyStatement()
-        publish_permissions.add_arn_principal(f'arn:aws:iam::{Aws.ACCOUNT_ID}:root')
-        for id in account_ids:
-            subscribe_permissions.add_arn_principal(f'arn:aws:iam::{id}:root')
+        publish_permissions.add_arn_principal(f"arn:aws:iam::{Aws.ACCOUNT_ID}:root")
+        for id in all_language_account_ids:
+            subscribe_permissions.add_arn_principal(f"arn:aws:iam::{id}:root")
         publish_permissions.add_actions("SNS:Publish")
         publish_permissions.add_service_principal("events.amazonaws.com")
         publish_permissions.add_resources(topic.topic_arn)
         topic.add_to_resource_policy(publish_permissions)
+
+        #####################################
```

Comment:
Advisory: I don't personally like large manually formatted comment blocks like this? They are hard to maintain, and are visually jarring in flow. I would refactor this to a method `init_logging_bucket(self)` (and below, either part of that method, or a new method `init_logging_bucket_role`

Reply:
Done.